spark/src/main/java/org/apache/iceberg/spark/source/DeleteRowReader.java

```diff
@@ -52,6 +56,10 @@ public EqualityDeleteRowReader(CombinedScanTask task, Schema schema, Schema expe
     // update the current file for Spark's filename() function
     InputFileBlockHolder.set(file.path().toString(), task.start(), task.length());
 
-    return matches.findEqualityDeleteRows(open(task, requiredSchema, idToConstant)).iterator();
+    if (deleteContent.equals(FileContent.EQUALITY_DELETES)) {
```

Comment:
I think that this should use a combined `keepDeletedRows` method instead of methods specific to equality or position deletes.